### PR TITLE
Test-Connection - Remove informational output, consolidate Ping usage

### LINKED ISSUE #6768: Test-Connection cmdlet displaying unwanted data.
Steps to reproduce
------------------

```powershell
The Test-Connection cmdlet is displaying unwanted data as part of the result.

Code:

Test-Connection www.microsoft.com -Count 1 -Quiet
```

Expected behavior
-----------------

```none
It should display just the word: True
```

Actual behavior
---------------

```none
Pinging www.microsoft.com [23.204.153.19] with 32 bytes of data:
Reply from 23.204.153.19: bytes=32 time=17ms TTL=58
Ping complete.
True
```

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0-preview.2
PSEdition                      Core
GitCommitId                    v6.1.0-preview.2
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### PR DESCRIPTION
**Pull request by vexx32**
# PR Summary

1. ~~Replaces all `WriteInformation()` calls with `WriteVerbose()`~~ **Removes** all verbose and progress output. We will revisit verbose output in a later PR.
2. Refactors ping implementation to create only a single `Ping` object which is reused for all `Send()` calls.
3. Implements `IDisposable` for `Test-Connection` so that it can dispose of the `Ping` object when it is no longer needed.

## PR Context

Part 2 of #10044, refactor of `Test-Connection` cmdlet.
Fixes #6768

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by iSazonov** [comment]
> Replaces all WriteInformation() calls with WriteVerbose()

Intention was to write to Host. PowerShell implements this over information stream. 
A conclusion was to be close to Windows PowerShell so I see no reason to write to verbose stream in the same way. I believe we need to remove this.

**Comment by vexx32** [comment]
That is true...

I was thinking of a use case where `-Quiet` may be used in the code, but during development the `-Verbose` parameter could be added to show the otherwise hidden/ignored information.

Though I suppose the verbose output probably should just be something along the lines of: `VERBOSE: Testing connection to <target>`

**Comment by iSazonov** [comment]
Yes, verbose output should has another form and probably be in other places because it serves other purposes. We could implement this later if this will be needed.

**Comment by vexx32** [comment]
I'll do a simple revision of the verbose output today. We can expand on it more at a later date if we need more. :)

**Comment by iSazonov** [comment]
You could simply remove the extra output in the PR and add verbose in follow PR to speed up code review.

**Comment by vexx32** [comment]
@iSazonov would it be better to remove progress bars as well in this PR or following PR?

File: src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ namespace Microsoft.PowerShell.Commands
     [OutputType(typeof(bool), ParameterSetName = new string[] { DefaultPingParameterSet, RepeatPingParameterSet, TcpPortParameterSet })]
```
Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs]: @vexx32, your last commit had 1 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/cmdlet-example/command-line-simple-example.md.https://www.microsoft.com/net/core#linuxubuntu should work
```powershell
retry of URL failed with error: Response status code does not indicate success: 400 (Bad Request).
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 117
117:                                     throw "retry of URL failed with error: $($_.Exception.Message)"

```

**Comment by vexx32** [comment]
@PoshChan please retry static

File: src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs [hunk 2]
```diff
@@ -960,12 +782,18 @@ private byte[] GetSendBuffer(int bufferSize)
         private const int DefaultSendBufferSize = 32;
         private static byte[] s_DefaultSendBuffer = null;
 
-        // Random value for WriteProgress Activity Id.
-        private static readonly int s_ProgressId = 174593053;
+        private bool disposed;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs]: nit: _disposed

**Review by iSazonov — APPROVED** [review]

**Comment by vexx32** [comment]
@TravisEz13 is there anything more you guys need from me on this one? I have a couple more PRs lined up and ready for this one that I'd love to get in for PSv7 to more fully cover the scope of the approved RFC. 💖

**Comment by vexx32** [comment]
@SteveL-MSFT @TravisEz13 I know you guys are pretty busy this timeframe for release, but can't hurt to ask I hope:

Can we get this one merged in and move forward with implementing the remainder of the changes already approved in the RFC for v7 timeframe? I have the code for it all written and the RFC is already approved, so I would like to have all the Test-Connection changes I plan on making merged in for v7 GA if at all possible.

Thank you! You guys are awesome! 😊 💖

**Comment by TravisEz13** [comment]
Did the output of the Cmdlet change in this PR?

**Comment by vexx32** [comment]
@TravisEz13 The success output of this cmdlet has not changed. That is what I plan to change in a follow-up PR. 🙂 

Or, if you'd prefer, I can merge the changes from that branch into here directly, but that does make the diffs unfortunately difficult to work through, which is why I subdivided my PRs on this one. 🙂

**Comment by TravisEz13** [comment]
No... I prefer it like this.  I'm just trying to clarify.

**Pull request merged**